samples/net/dsa/src/dsa_lldp.c

```diff
@@ -18,24 +18,89 @@
 #include <errno.h>
 
 #include <zephyr/logging/log.h>
-#include "main.h"
 
 /* Loglevel of dsa_lldp function */
-LOG_MODULE_DECLARE(net_dsa_lldp_sample, CONFIG_NET_DSA_LOG_LEVEL);
+LOG_MODULE_DECLARE(net_dsa_sample, CONFIG_NET_DSA_LOG_LEVEL);
 
-#define LLDP_SYSTEM_NAME_SIZE 24
-#define LLDP_ETHER_TYPE 0x88CC
+#include "dsa_lldp.h"
+
+#define LLDP_SYSTEM_NAME_SIZE    24
+#define LLDP_ETHER_TYPE          0x88CC
 #define LLDP_INPUT_DATA_BUF_SIZE 512
-#define DSA_BUF_SIZ 128
-int dsa_lldp_send(struct net_if *iface, struct instance_data *pd,
-		  uint16_t lan, int src_port, int origin_port, int cmd,
-		  struct eth_addr *origin_addr)
+#define DSA_BUF_SIZ              128
+
+static const uint8_t eth_filter_l2_addr_base[][6] = {
+	/* MAC address of other device - for filtering testing */
+	{0x01, 0x80, 0xc2, 0x00, 0x00, 0x03}};
+
+enum net_verdict dsa_ll_addr_switch_cb(struct net_if *iface, struct net_pkt *pkt)
+{
+	struct net_eth_hdr *hdr = NET_ETH_HDR(pkt);
+	struct net_linkaddr lladst;
+
+	net_pkt_cursor_init(pkt);
+	lladst.len = sizeof(hdr->dst.addr);
+	lladst.addr = &hdr->dst.addr[0];
+
+	/*
+	 * Pass packet to lan1..3 when matching one from
+	 * check_ll_ether_addr table
+	 */
+	if (check_ll_ether_addr(lladst.addr, &eth_filter_l2_addr_base[0][0])) {
+		return 1;
```

Comment:
Please do not return "plain" values here and in the final return, but either `NET_OK`, `NET_CONTINUE` or `NET_DROP`

Reply:
okay